### ISSUE TITLE
Mock how Next.js builds to test client/lib/documentation.ts

### ISSUE THREAD

**Issue by joulev** [joulev/ezkomment#56]
Currently `getFileData()` in `@client/lib/documentation.ts` uses `fetch` (not from anywhere?) and `readFileSync` from `fs` at the same time.

On browsers, `readFileSync` is not available; meanwhile on Node.js v16 `fetch` is not [yet](https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#fetch) available. Therefore the function cannot run on any Jest environment that I know. As a result I am unable to test that file (yet).

In the app, the function is run during the build process (used inside [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching/get-static-props)). Since (somehow) `fetch` is available there, it works smoothly. But that means in order to even run `getFileData()` in a test file, it's necessary to replicate the Next.js build environment, which I haven't been able to do. Google also doesn't help much.

I can probably circumvent this by using a library to fetch data in a Node.js environment (`node-fetch`, `axios` or whatever), then mock both the fetching function and `fs` in the test file. But that means I'm installing a few extra dependencies just for one test file, and my app code will unnecessarily get a bit more complex. So I am not going to do that.

**tl;dr** how can I mock the Next.js building environment, where magically `fetch` is available as a global object and `fs` is also available?

**Comment by joulev** [comment]
Turns out it is not that difficult. We have to mock the response anyway, so mocking `global.fetch` in Node environment is not that hard to do.

**Comment by joulev** [comment]
Further information: Next.js actually polyfills `fetch` (of course, there's no way `fetch` just automagically shows up), [documentation](https://nextjs.org/docs/basic-features/supported-browsers-features#polyfills).
